src/engine/CodeValidator.v3

```diff
@@ -1266,6 +1351,15 @@ class CodeValidator(extensions: Extension.set, limits: Limits, module: Module, e
 			if (!ValueTypes.isAssignable(ct[j], p[j])) err_atpc().TypeMismatchInCatch(i, p[i], ct[i]);
 		}
 	}
+	def checkContHandle(i: int, tag: TagDecl, target: ControlEntry) {
+		var p = labelArgs(target);
+		if (p == null) return;
+		var ct = Arrays.append(ValueType.Ref(true, HeapType.ANY), tag.fields);
+		if (ct.length != p.length) return err_atpc().ArityMismatchInCatch(i, p.length, ct.length);
+		for (j < p.length) {
+			if (!ValueTypes.isAssignable(p[j], ct[j])) err_atpc().TypeMismatchInCatch(i, p[i], ct[i]);
```

Comment:
Let's update the error method to `TypeMismatchInHandler` and have it take an argument for "catch" (i.e. catch handler) and "resume" (i.e. resume handler).